ui/src/shared/components/ViewOverlay.tsx

```diff
@@ -0,0 +1,106 @@
+// Libraries
+import React, {PureComponent} from 'react'
+
+// Components
+import {Button, SpinnerContainer, TechnoSpinner} from '@influxdata/clockface'
+import {Overlay, ComponentSize} from 'src/clockface'
+import {Controlled as ReactCodeMirror} from 'react-codemirror2'
+import CopyButton from 'src/shared/components/CopyButton'
+
+// Types
+import {DocumentCreate} from '@influxdata/influx'
+import {ComponentColor} from '@influxdata/clockface'
+import {RemoteDataState} from 'src/types'
+
+interface Props {
+  onDismissOverlay: () => void
+  resource: DocumentCreate
+  overlayHeading: string
+  status: RemoteDataState
+  isVisible: boolean
+}
+
+export default class ViewOverlay extends PureComponent<Props> {
```

Comment:
do we need to add the errors decorator here?

Reply:
I updated it!